### PR TITLE
Code restructuring to support V1alpha1 and V1alpha2 API

### PR DESCRIPTION
**Pull request by johnugeorge**
Code is restructured to support v1alpha1 and v1alpha2 api. Docker images of each component contains v1alpha1 and v1alpha2  binaries(Currently  added only  for `controller` and `db` components).  Default is still v1alpha1 binary.

1. v1alpha2 Controller and DB  Bolierplate code is added. I will add the initial implementation  in a separate PR
2. v1alpha2 version of  Suggestion is not added yet.  Implementation can be merged with #407 which tracks the python migration of algorithms
3. GRPC API Boilerplate structure is added.   Related: #431. 
4. v1alpha2 test structure is not added. It can be added once all v1alpha2 components are created. 

Related: #670

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/448)

### CODE REVIEW

**Comment by johnugeorge** [comment]
@richardsliu @YujiOshima @hougangliu @gaocegege  @jdplatt @achalshant

**Comment by richardsliu** [comment]
/lgtm
Thanks!

File: cmd/studyjobcontroller/v1alpha2/main.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/kubeflow/katib/pkg/api/operators/apis"
+	controller "github.com/kubeflow/katib/pkg/controller/v1alpha2"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+)
+
+func main() {
```
Comment (andreyvelich) [inline cmd/studyjobcontroller/v1alpha2/main.go]: Will Katib controller have hame "StudyJobController" in v1alpha2?

**Comment by richardsliu** [comment]
For the NAS files - we should make sure their naming follow the conventions of the rest of the repo. It can be fixed later.

**Comment by andreyvelich** [comment]
@johnugeorge Thank you for your PR!

**Comment by andreyvelich** [comment]
@johnugeorge Should we also put these manifests https://github.com/kubeflow/katib/tree/master/manifests in v1alpha1 folder?

**Comment by hougangliu** [comment]
/lgtm
thanks!

**Comment by YujiOshima** [comment]
/lgtm
Thank you!
As @andreyvelich pointed, we should think about the folder name of `cmd/studyjobcontroller` later.

File: cmd/studyjobcontroller/v1alpha2/main.go (hunk 1 above)
Comment (johnugeorge) [reply cmd/studyjobcontroller/v1alpha2/main.go]: updated

**Comment by johnugeorge** [comment]
renamed `studyjobcontroller` to `katib-controller` 

Image name is not changed yet. It will be changed later when all v1alpha2 components are built.`vizier` term in image names will also be renamed to `katib`

I will raise a separate PR for separating v1alpha1 in `manifests`, `scripts`, `tests`  folders

**Comment by YujiOshima** [comment]
/lgtm 
Thanks!

**Pull request merged**